File: simple/pythonvulnweb/code_exec/interface_impl.py
from interface.vuln_interface import VulnInterface
from common.vuln_parent import VulnParent

class CodeExecInterfaceImpl(VulnInterface, VulnParent):
    def execute(self, code):
        # 代码执行漏洞测试：接口实现类，继承父类并实现抽象方法
        # 调用链：外部参数code -> execute方法 -> 父类code_exec -> eval
        # 漏洞点：eval执行用户控制的code参数
        return self.code_exec(code)
    
    def hardcoded_execute(self):
        # 代码执行漏洞测试：接口实现类，硬编码场景
        # 调用链：hardcoded_execute方法 -> 父类hardcoded_code_exec -> eval
        return self.hardcoded_code_exec()

File: simple/phpvulnweb/code_exec/InterfaceImpl.php
<?php
// 代码执行 - 接口实现类调用测试文件
// 调用链：外部请求参数(code) → InterfaceImpl.php → VulnInterface 实现类 → VulnParent.execCode

require_once __DIR__ . '/../common/VulnParent.php';
require_once __DIR__ . '/../interface/VulnInterface.php';

// 代码执行接口实现类，同时继承 VulnParent，实现「接口 + 类继承」组合调用
class CodeExecImpl extends VulnParent implements VulnInterface
{
    // 实现接口 execute 方法：接收外部可控参数 $payload 触发代码执行
    public function execute($payload)
    {
        $this->execCode($payload);
    }

    // 实现接口 executeFixed 方法：内置固定参数触发代码执行
    public function executeFixed()
    {
        $this->execCodeHardcoded();
    }
}
?>
<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <title>Code Execution - Interface Implementation Test</title>
</head>
<body>
    <h1>Code Execution - Interface Implementation Test</h1>
    <p>Usage: ?code=phpinfo();</p>
    <hr>
    <?php
    $impl = new CodeExecImpl();
    if (isset($_GET['code']) && !empty($_GET['code'])) {
        // 调用链：InterfaceImpl.php → CodeExecImpl.execute → VulnParent.execCode（漏洞点）
        $impl->execute($_GET['code']);
    } else {
        echo "<p>No code parameter provided. Running hardcoded test...</p>";
    }
    $impl->executeFixed();
    ?>
</body>
</html>
